File: regions/system/library/typescript/server-common/src/error-detail.ts
/**
 * エラーの詳細情報を表すインターフェース。
 * REST-API設計 D-007 仕様に準拠したフィールドレベルのエラー情報を提供する。
 */
export interface ErrorDetail {
  /** エラーが発生したフィールド名 */
  field: string;
  /** エラーの理由（機械可読） */
  reason: string;
  /** 人間向けのエラーメッセージ */
  message: string;
}

File: regions/system/library/dart/server_common/lib/src/error_detail.dart
/// エラー詳細情報。
/// フィールドレベルのバリデーションエラー等の追加コンテキストを提供する。
/// REST-API設計.md D-007 仕様に準拠する。
class ErrorDetail {
  /// エラーが発生したフィールド名
  final String field;

  /// エラーの理由（機械可読）
  final String reason;

  /// エラーメッセージ（人間可読）
  final String message;

  const ErrorDetail({
    required this.field,
    required this.reason,
    required this.message,
  });

  /// JSON マップに変換する。
  Map<String, dynamic> toJson() => {
        'field': field,
        'reason': reason,
        'message': message,
      };
}
